File: Leetcode/905/Selection_Sort_Algo/main.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            if (nums[j] % 2 == 0) {
                // swap
                let temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                break;
            }
        }
    }
    return nums;
    
};

File: Leetcode/905/Selection_Sort_Algo/Main.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int n = nums.length;
        
        for (int i = 0; i < n; i++) {
            int iseven = -1;

            for (int j = i + 1; j < n; j++) {
                if (nums[j] % 2 == 0) {
                    iseven = j;
                    break;
                }
            }

            if (iseven != -1) {
                // swap
                int temp = nums[i];
                nums[i] = nums[iseven];
                nums[iseven] = temp;
            }
        }

        return nums;
        
    }
}